### PR TITLE
Mesh-747 RPC calls for MIPS and Identity modules

### PR DESCRIPTION
**Pull request by vkandy**
Implemented following RPC calls

**1. Get vote count for a proposal identified by MIPS Index**
----
`mips_getVotes(index)` retrieves vote summary for a proposal given by MIPS index. 

**NOTE:** Stake is measured in `Balance` which is capped (or expanded) to `u64` to avoid serde issues with `u128`.

1. When a proposal doesn't exist `proposalNotFound` is returned:
`$ curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mips_getVotes", "params": [0]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":"ProposalNotFound","id":1}`

2. When a proposal is found by MIPS index:
`$ curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mips_getVotes", "params": [0]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":{"success":{"ayes":50000000000,"nays":2560000000}},"id":1}`

**2. Proposals created by an AccountId**
----
`mips_proposedBy(address)` retrieves indicies of proposals started by `address`

1. Empty array in response when given address hasn't created any proposals
`$ curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mips_proposedBy", "params": ["5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":[],"id":1}`

2. Alice creates two proposals
`$ curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mips_proposedBy", "params": ["5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":[0,1],"id":1}`

**3. Proposals an account voted on**
----
`mips_votedOn` retrieves proposals an account voted on.

1. Bob hasn't voted on any proposals
`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mips_proposedBy", "params": ["5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":[],"id":1}`

2. Bob voted on a proposal Alice created earlier
`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "mips_votedOn", "params": ["5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":[0],"id":1}`

**4. Retrieve all master / signing keys associated with the IdentityId**
----
1. When IdentityId is not found
`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "identity_getDidRecords", "params": [<IdentityId>]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":"idNotFound","id":1}`

2. For a valid IdentityId
`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "identity_getDidRecords", "params": [<Alice Id>]}' http://localhost:9933`
`{"jsonrpc":"2.0","result":{"success":{"master_key":[212,53,147,199,21,253,211,28,97,20,26,189,4,169,159,214,130,44,133,88,133,76,205,227,154,86,132,231,165,109,162,125],"signing_items":[]}},"id":1}`

### CODE REVIEW

File: pallets/mips/src/lib.rs [hunk 1]
```diff
@@ -465,6 +475,57 @@ impl<T: Trait> Module<T> {
             .collect()
     }
 
+    /// Retrieve votes for a proposal represented by MipsIndex `index`.
+    pub fn get_votes(index: MipsIndex) -> VoteCount<BalanceOf<T>>
```
Comment (satyamakgec) [inline pallets/mips/src/lib.rs]: I suggest having different `impl` block for getters. Part of the best practice to segregate the state transition function/used internally and external getters.

File: pallets/mips/src/lib.rs [hunk 2]
```diff
@@ -465,6 +475,57 @@ impl<T: Trait> Module<T> {
...
+            VoteCount::ProposalNotFound
+        }
+    }
+
+    /// Retrieve proposals made by `address`.
+    pub fn proposed_by(address: T::AccountId) -> Vec<MipsIndex> {
+        Self::proposal_meta()
+            .into_iter()
+            .filter(|meta| meta.proposer == address)
+            .map(|meta| meta.index)
```
Comment (satyamakgec) [inline pallets/mips/src/lib.rs]: you can use `filter_map()` here.

File: pallets/mips/rpc/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+/// Error type of this RPC api.
+pub enum Error {
+    /// The transaction was not decodable.
+    DecodeError,
+    /// The call to runtime failed.
+    RuntimeError,
+}
+
+impl From<Error> for i64 {
```
[the remark this answers is not in the document]
Comment (vkandy) [reply pallets/mips/rpc/src/lib.rs]: Instead, I've removed `Error::RuntimeError as i64` and used enum variant of `Error` to capture specific codes.

File: pallets/mips/rpc/runtime-api/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+version = "2.0.0"
+authors = ["Anonymous"]
+edition = "2018"
+
+[dependencies]
+serde = { version = "1.0.101", optional = true, features = ["derive"] }
+codec = { package = "parity-scale-codec", version = "1.0.0", default-features = false, features = ["derive"] }
+sp-api = { rev = '93569d0d1af7aa5eb67b9de8b58d7fffde63b421', git = 'https://github.com/paritytech/substrate.git', default_features = false}
+sp-std = { rev = '93569d0d1af7aa5eb67b9de8b58d7fffde63b421', git = 'https://github.com/paritytech/substrate.git', default_features = false }
+sp-runtime = { rev = '93569d0d1af7aa5eb67b9de8b58d7fffde63b421', git = 'https://github.com/paritytech/substrate.git', default_features = false }
```
Comment (satyamakgec) [inline pallets/mips/rpc/runtime-api/Cargo.toml]: Could you please reverse the order here as well in all of the above packages

File: pallets/identity/rpc/src/lib.rs [hunk 5]
```diff
@@ -53,16 +60,28 @@ pub enum Error {
     RuntimeError,
 }
 
-impl<C, Block, IdentityId, Ticker> IdentityApi<<Block as BlockT>::Hash, IdentityId, Ticker>
+impl From<Error> for i64 {
```
Comment (fmiguelgarcia) [inline pallets/identity/rpc/src/lib.rs]: Enum can be cast as integers (i32): see [Enums C-like](https://doc.rust-lang.org/rust-by-example/custom_types/enum/c_like.html)

File: pallets/mips/src/lib.rs [hunk 6]
```diff
@@ -457,11 +467,16 @@ decl_module! {
 
 impl<T: Trait> Module<T> {
     /// Retrieve all proposals that need to be closed as of block `n`.
-    pub fn proposals_maturing_at(n: T::BlockNumber) -> Vec<(MipsIndex, T::Hash)> {
+    pub fn proposals_maturing_at(block: T::BlockNumber) -> Vec<(MipsIndex, T::Hash)> {
         Self::proposal_meta()
             .into_iter()
-            .filter(|meta| meta.end == n)
-            .map(|meta| (meta.index, meta.proposal_hash))
+            .filter_map(|meta| {
```
Comment (fmiguelgarcia) [inline pallets/mips/src/lib.rs]: Previous version looks more readable (for me, it is purely personal opinion) and concise. 
It there any reason to this change that I miss?

**Pull request merged**